pkg/api-server/dataplane_layout_endpoint.go

```diff
@@ -0,0 +1,186 @@
+			Doc("Get Dataplane Layout").
+			Returns(http.StatusOK, "OK", nil),
+	)
+}
+
+func (dle *dataplaneLayoutEndpoint) getLayout(request *restful.Request, response *restful.Response) {
+	meshName := request.PathParameter("mesh")
+	dataplaneName := request.PathParameter("name")
+
+	if err := dle.resourceAccess.ValidateGet(
+		request.Request.Context(),
+		core_model.ResourceKey{Mesh: meshName, Name: dataplaneName},
+		core_mesh.DataplaneResourceTypeDescriptor,
+		user.FromCtx(request.Request.Context()),
+	); err != nil {
+		rest_errors.HandleError(request.Request.Context(), response, err, "Access Denied")
+		return
+	}
+
+	mesh, err := dle.getMeshResource(request.Request.Context(), meshName)
+	if err != nil {
+		rest_errors.HandleError(request.Request.Context(), response, err, "Failed to retrieve Mesh")
+		return
+	}
+
+	if mesh.Spec.GetMeshServices().GetMode() == v1alpha1.Mesh_MeshServices_Disabled {
+		response.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	dataplane, err := dle.getDataplaneResource(request.Request.Context(), meshName, dataplaneName)
+	if err != nil {
+		rest_errors.HandleError(request.Request.Context(), response, err, "Failed to retrieve Dataplane")
+		return
+	}
+
+	baseMeshContext, err := dle.meshContextBuilder.BuildBaseMeshContextIfChanged(request.Request.Context(), meshName, nil)
+	if err != nil {
+		rest_errors.HandleError(request.Request.Context(), response, err, "Failed to build MeshContext")
+	}
+
+	inbounds := util_slices.Map(dataplane.Spec.GetNetworking().GetInbound(), func(inbound *v1alpha1.Dataplane_Networking_Inbound) api_common.DataplaneInbound {
+		sectionName := inbound.GetName()
+		if sectionName == "" {
+			sectionName = strconv.Itoa(int(inbound.GetPort()))
+		}
+
+		return api_common.DataplaneInbound{
+			Kri:      kri.From(dataplane, sectionName).String(),
+			Port:     int32(inbound.GetPort()),
+			Protocol: inbound.GetProtocol(),
+		}
+	})
+
+	var outbounds []api_common.DataplaneOutbound
+	if dataplane.Spec.GetNetworking().GetOutbound() != nil {
+		// TODO handle user defined outbounds on universal without transparent proxy
+	} else {
+		// handle reachable backend and outbounds when using transparent proxy
+		reachableBackends := baseMeshContext.DestinationIndex.GetReachableBackends(mesh, dataplane)
+		outbounds = dle.getOutbounds(baseMeshContext.DestinationIndex, reachableBackends)
+	}
+
+	slices.SortStableFunc(outbounds, func(a, b api_common.DataplaneOutbound) int {
+		return strings.Compare(a.Kri, b.Kri)
+	})
+
+	networkingLayout := types.DataplaneNetworkingLayout{
+		Inbounds:  inbounds,
+		Kri:       kri.From(dataplane, "").String(),
+		Labels:    dataplane.GetMeta().GetLabels(),
+		Outbounds: outbounds,
+	}
+
+	err = response.WriteHeaderAndJson(http.StatusOK, networkingLayout, "application/json")
+	if err != nil {
+		log.Error(err, "Could not write response")
+	}
+}
+
+func (dle *dataplaneLayoutEndpoint) getMeshResource(ctx context.Context, meshName string) (*core_mesh.MeshResource, error) {
```

Comment:
I don't find these functions super useful. They add double error handling and hide things that might actually not need to be hidden...

Reply:
removed